pkg/morph/client/reload.go

```diff
@@ -0,0 +1,27 @@
+package client
+
+import "slices"
+
+// Reload allows runtime reconfiguration for WithEndpoints parameter.
+//
+// Keeping the current connection alive, if in the config there is an endpoint that the client is connected to.
+func (c *Client) Reload(opts ...Option) {
+	cfg := new(cfg)
+	for _, o := range opts {
+		o(cfg)
+	}
+
+	c.cfg.endpoints = cfg.endpoints
+
+	c.endpoints = cfg.endpoints
```

Comment:
This needs to be protected wrt other threads, btw.

Reply:
[Here](https://github.com/nspcc-dev/neofs-node/pull/2996/commits/5ec3d5a1762e3dabc287fdd0d535306c9b38d278#) `switchLock *sync.RWMutex` has been deleted. Do I need to add it again or is there some other way?

Reply:
It was created for a different purpose. You can introduce som endpoints RWLock now.